derive_builder_core/src/error.rs

```diff
@@ -0,0 +1,32 @@
+use std::{error::Error, fmt};
+
+/// Runtime error when a `build()` method is called and one or more required fields
+/// do not have a value.
+#[derive(Debug, Clone)]
+pub struct UninitializedFieldError(&'static str);
+
+impl UninitializedFieldError {
+    /// Create a new `UnitializedFieldError` for the specified field name.
+    pub fn new(field_name: &'static str) -> Self {
+        UninitializedFieldError(field_name)
+    }
+
+    /// Get the name of the first-declared field that wasn't initialized
+    pub fn field_name(&self) -> &'static str {
+        self.0
+    }
+}
+
+impl fmt::Display for UninitializedFieldError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Field not initialized: {}", self.0)
+    }
+}
+
+impl Error for UninitializedFieldError {}
+
+impl From<&'static str> for UninitializedFieldError {
```

Comment:
Is this really needed?

Reply:
I suppose not; is there a reason to exclude it? I generally include `From` conversions when `new` takes a single argument, but the macro could invoke `new` directly.

Reply:
Same concerns as in #192. `&str` is too broad.

Reply:
In this case there isn't ambiguity though; we know that we intended to represent an uninitialized field, as opposed to a custom validation error. The macro-generated code uses `UninitializedFieldError::from` since `.into().into()` almost never works in Rust.

Reply:
I've confirmed that there is no `From<&'static str>` direct to the generated builder error, only to the `UninitializedFieldError`.